### DISCUSSION TITLE
Is there a way to use a custom DataCollection class?

### DISCUSSION THREAD

**Discussion by gigerIT** [spatie/laravel-data#396] [Q&A]
Hi there
I'd like to know if the package supports a way of extending the DataCollection class or adding macros to it to extend the functions available on the collection class.

The docs state 
> The DataCollection class implements a few of the Laravel collection methods:

but we have a project where we need more than those few functions.

Just typehinting to our own collection class like this:
```php
#[DataCollectionOf(SongData::class)]
public CustomDataCollection $songs;
```
Dosen't work, as:
```php
SongData::collection(Song::all());
```
always seems to create a default DataCollection instance for it.

**Comment by rubenvanassche** [comment]
I've added some docs on this: https://github.com/spatie/laravel-data/blob/0f1de38df97ccd9cc96f6244255136e61dc09b84/docs/custom-collections.md#L1-L0
